### ISSUE TITLE
Package: go-neb: it's been discontinued

### ISSUE THREAD

**Issue by herbetom** [NixOS/nixpkgs#342663]
Not sure if there is some policy around this topic (haven't found it or an template) so going a bit free style.

- Package name: [go-neb](https://github.com/NixOS/nixpkgs/blob/261a20d9080cfc1bd7730c6ab1e1d6491d68cf15/pkgs/applications/networking/instant-messengers/go-neb/default.nix)
- Archived: [2023-06-23](https://github.com/matrix-org/go-neb/commit/3ba3d4600f42d913824ae36cf98d113d6a0d1099)

The go-neb project was archived/discontinued by the developer last year. Currently it seems to work still but with it beeing abandoned sofware i think a removal/marking as insecure or whatever should be considered at some point.

It doesn't look like there are viable forks to switch to. Or i haven't quickly found them.

On a side node there is at least one [prominent](https://github.com/NixOS/infra/blob/51f10849692e56914cd5f344e6fd38812801f14c/build/pluto/prometheus/alertmanager.nix#L75-L135) deployment.

**Notify maintainers**
@mweinelt @maralorn

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by maralorn** [comment]
I am not neccessarily against a removal but I am currently using this very successfully for alertmanager notifications and would appreciate finding an alternative before we drop it. 😆

**Comment by herbetom** [comment]
Yes, im also still looking into alternatives exactly for those notifcations and will have to migrate one deployment. :sweat_smile:

There is [matrix-hook](https://github.com/pinpox/matrix-hook) ([nixpkgs](https://github.com/NixOS/nixpkgs/blob/95865f4ea75d70bdcb2a3733c58b7fc564cb87be/pkgs/by-name/ma/matrix-hook/package.nix)) as seen in [nix-community/infra](https://github.com/nix-community/infra/blob/1bbe18abd8e04053bf14ddfacd78653c14227f10/modules/nixos/monitoring/matrix-hook.nix#L14-L36). It has fewer dependencies but it also doen't look too actively developed (the dependencies are around a year old (the last go-neb bump was ~two years ago so it's newer)).


Then there is the in the discontinuation announcement mentioned [matrix-hookshot](https://github.com/matrix-org/matrix-hookshot) ([nixpkgs](https://github.com/NixOS/nixpkgs/blob/1b8c065923d84726d8c9bbbdf7842fad9e05aab1/pkgs/servers/matrix-synapse/matrix-hookshot/default.nix)). It covers more of what go-neb does/did and apparently also has [some support for alertmanager](https://github.com/matrix-org/matrix-hookshot/blob/2763801d1b677a5a66ee5f1a3b822772c50fce11/contrib/jsTransformationFunctions/alertmanager.js).

Don't know if there are other alternatives.

**Comment by Erethon** [comment]
[Matrix-alertmanager](https://github.com/jaywink/matrix-alertmanager) exists specifically for alertmanager based notifications. Hookshot is the "official" replacement for go-neb for things outside of alertmanager notifications.

**Comment by herbetom** [comment]
Just noticed that matrix-hookshot utiziles the [Matrix Application Services](https://element-hq.github.io/synapse/latest/application_services.html).
And that is something which you apparently have to configure in your homeserver config. So to run such a hookshot bot you basically need a matrix homeserver under your administrative control and can't just use some random matrix User account on some third party homeserver. Might make sense from a feature perspective. But that's quite a barrier. :/

**Comment by maralorn** [comment]
I had migrated my setup to matrix-altermanager and that worked fine.

**Comment by herbetom** [comment]
I don't think that's packaged yet? Would be kinda great if that would happen prior to removal ... Also a module would be kinda nice 😅

@marcin-serwin In my opinion increasing pressure by marking go-neb as insecure might be a good immediate next step. But i won't rule out that this could be colored by myself also still having to do a migration to somewhere else 🙈

**Comment by maralorn** [comment]
Its packaged and there is a module: https://search.nixos.org/options?channel=25.11&query=matrix-alertmanager

I decided to go for a different solution because of different requirements, but it worked fine until this removal commit for me:

https://code.maralorn.de/maralorn/config/commit/0089fca6a7cf99cb7f4b786a3ffe20cab15d221e

**Comment by mweinelt** [comment]
Removing nixos/infra as a blocker.

https://github.com/NixOS/infra/pull/1021

**Comment by siraben** [comment]
Closed by #559393
